api/message.go

```diff
@@ -331,8 +336,37 @@ func (c *Client) DeleteMessage(channelID discord.ChannelID, messageID discord.Me
 // any message provided is older than that or if any duplicate message IDs are
 // provided.
 //
+// Because the underlying endpoint only supports a maximum of 100 message IDs
+// per request, DeleteMessages will make a total of messageIDs/100 rounded up
+// requests.
+//
 // Fires a Message Delete Bulk Gateway event.
 func (c *Client) DeleteMessages(channelID discord.ChannelID, messageIDs []discord.MessageID) error {
+	switch {
+	case len(messageIDs) == 0:
+		return nil
+	case len(messageIDs) == 1:
+		return c.DeleteMessage(channelID, messageIDs[0])
+	case len(messageIDs) <= maxMessageDeleteLimit: // Fast path
+		return c.deleteMessagesRaw(channelID, messageIDs)
+	}
+
+	for start := 0; start < len(messageIDs); start += 100 {
+		end := start + maxMessageDeleteLimit
```

Comment:
We should consider just making a `min(i, j int)` function for all these abstracted functions.

~~tfw no generics~~